src/cryptonote_core/tx_pool.cpp

```diff
@@ -1073,7 +1073,7 @@ namespace cryptonote
 
     // If the total weight is too high, choose the best paying transactions
     if (total_weight > max_backlog_weight)
-      std::sort(tmp.begin(), tmp.end(), [](const auto& a, const auto& b){ return a.fee * b.weight > b.fee * a.weight; });
+      std::stable_sort(tmp.begin(), tmp.end(), [](const auto& a, const auto& b){ return a.fee * b.weight > b.fee * a.weight; });
```

Comment:
Isn't this formula wrong anyways? Shouldn't it be `x.fee / x.weight` instead of `*`? Right now, it's maximizing fee times weight, not fee per weight.